src/lsm/tree.zig

```diff
@@ -325,19 +312,24 @@ pub fn TreeType(comptime TreeTable: type, comptime Storage: type) type {
             key: Key,
             fingerprint: Fingerprint,
         ) LookupMemoryResult {
-            if (snapshot >= tree.lookup_snapshot_max.?) {
+            assert(snapshot <= snapshot_latest);
+
+            if (snapshot == snapshot_latest) {
```

Comment:
Overall, getting rid of `Tree.lookup_snapshot_max` makes a lot of sense to me, but I think I'm missing some context here.

With this PR, the mutable table is only visible when `snapshot_max` is passed. How are we going to handle cases where trees had different `lookup_snapshot_max` (eg IdTree and ObjectTree not sharing the same `lookup_snapshot_max`), but the snapshot provided could be found in both (eg `snapshot != snapshot_max and snapshot >= tree.lookup_snapshot_max.?`)?

Reply:
> * By `snapshot_max` do you mean `snapshot_latest`?

Yes, I fixed the typo 😳

> * `lookup_snapshot_max` previously had the invariant "Between beats, all of a groove's trees share the same lookup_snapshot_max." -- I'm not sure what you mean by "eg IdTree and ObjectTree not sharing the same...".

Exactly that invariant, if "between beats, all of a groove's trees share the same lookup_snapshot_max.", So I suppose it can diverge at some point. But I'm not sure if we can do lookups during this state (that's the context I might be missing).

> This is a bit speculative since querying snapshots isn't implemented. Note the comment in `tree.zig`:

Got it :+1:

Reply:
> But I'm not sure if we can do lookups during this state (that's the context I might be missing).

We did perform lookups in this state, which is why each `Tree` had its own `lookup_snapshot_max`, rather than sharing a single snapshot between all `Tree`.

But since we now only hit the mutable table when `snapshot == snapshot_latest` that shouldn't be necessary.

---

For a little more context: The reason we needed `lookup_snapshot_max` in the first place is that (some weeks/months ago) compaction's output tables were added to the manifest before the tables' blocks were written to disk. But as part of an unrelated change that is no longer the case -- we only update the manifest _after_ the corresponding writes are finished. 

`lookup_snapshot_max`'s job was to guard those un-written tables from being read, so it isn't necessary anymore.

Reply:
Thanks for the explanation @sentientwaffle 🎉